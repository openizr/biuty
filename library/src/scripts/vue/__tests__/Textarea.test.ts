/**
 * Copyright (c) Matthieu Jabbour. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { mount } from '@vue/test-utils';
import UITextarea from 'scripts/vue/Textarea.vue';

type component = any; // eslint-disable-line @typescript-eslint/no-explicit-any

jest.mock('scripts/helpers/generateRandomId');

describe('vue/UITextarea', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly - basic', () => {
    const wrapper = mount(UITextarea, {
      propsData: { name: 'test', modifiers: 'large' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with id', () => {
    const wrapper = mount(UITextarea, {
      propsData: { name: 'test', id: 'test' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with cols and rows', () => {
    const wrapper = mount(UITextarea, {
      propsData: {
        name: 'test', cols: 10, rows: 50,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with placeholder', () => {
    const wrapper = mount(UITextarea, {
      propsData: { name: 'test', placeholder: 'text' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with helper', () => {
    const wrapper = mount(UITextarea, {
      propsData: { name: 'test', helper: 'Text' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with label', () => {
    const wrapper = mount(UITextarea, {
      propsData: { name: 'test', label: 'Text' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with value', async () => {
    const wrapper = mount(UITextarea, {
      propsData: { name: 'test', value: 'test' },
    });
    await wrapper.setProps({ value: 'new test' });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with maxlength', () => {
    const wrapper = mount(UITextarea, {
      propsData: { name: 'test', maxlength: 10 },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - disabled', () => {
    const wrapper = mount(UITextarea, {
      propsData: { name: 'test', modifiers: 'disabled' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - autocomplete off', () => {
    const wrapper = mount(UITextarea, {
      propsData: { name: 'test', autocomplete: 'off' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - readonly', () => {
    const wrapper = mount(UITextarea, {
      propsData: { name: 'test', readonly: true },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with listeners', async () => {
    const onFocus = jest.fn();
    const onBlur = jest.fn();
    const onChange = jest.fn();
    const wrapper = mount(UITextarea, {
      propsData: {
        name: 'test', value: 'test',
      },
      listeners: {
        focus: onFocus,
        blur: onBlur,
        change: onChange,
      },
    });
    await (wrapper.vm as component).focusField();
    await wrapper.find('textarea').setValue('new test');
    await wrapper.find('textarea').trigger('blur');
    expect(wrapper.html()).toMatchSnapshot();
    expect(onFocus).toHaveBeenCalledTimes(1);
    expect(onFocus).toHaveBeenCalledWith('test');
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith('new test');
    expect(onBlur).toHaveBeenCalledTimes(1);
    expect(onBlur).toHaveBeenCalledWith('new test');
  });
});
