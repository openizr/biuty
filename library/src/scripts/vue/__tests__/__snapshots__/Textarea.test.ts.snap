// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue/UITextarea renders correctly - autocomplete off 1`] = `
"<div class=\\"ui-textarea\\">
  <!---->
  <div class=\\"ui-textarea__wrapper\\"><textarea id=\\"z8a8d7a5zad4\\" name=\\"test\\" autocomplete=\\"off\\" class=\\"ui-textarea__wrapper__field\\"></textarea></div>
  <!---->
</div>"
`;

exports[`vue/UITextarea renders correctly - basic 1`] = `
"<div class=\\"ui-textarea ui-textarea--large\\">
  <!---->
  <div class=\\"ui-textarea__wrapper\\"><textarea id=\\"z8a8d7a5zad4\\" name=\\"test\\" autocomplete=\\"on\\" class=\\"ui-textarea__wrapper__field\\"></textarea></div>
  <!---->
</div>"
`;

exports[`vue/UITextarea renders correctly - disabled 1`] = `
"<div class=\\"ui-textarea ui-textarea--disabled\\">
  <!---->
  <div class=\\"ui-textarea__wrapper\\"><textarea id=\\"z8a8d7a5zad4\\" name=\\"test\\" autocomplete=\\"on\\" disabled=\\"disabled\\" class=\\"ui-textarea__wrapper__field\\"></textarea></div>
  <!---->
</div>"
`;

exports[`vue/UITextarea renders correctly - readonly 1`] = `
"<div class=\\"ui-textarea\\">
  <!---->
  <div class=\\"ui-textarea__wrapper\\"><textarea id=\\"z8a8d7a5zad4\\" name=\\"test\\" readonly=\\"readonly\\" autocomplete=\\"on\\" class=\\"ui-textarea__wrapper__field\\"></textarea></div>
  <!---->
</div>"
`;

exports[`vue/UITextarea renders correctly - with cols and rows 1`] = `
"<div class=\\"ui-textarea\\">
  <!---->
  <div class=\\"ui-textarea__wrapper\\"><textarea id=\\"z8a8d7a5zad4\\" name=\\"test\\" cols=\\"10\\" rows=\\"50\\" autocomplete=\\"on\\" class=\\"ui-textarea__wrapper__field\\"></textarea></div>
  <!---->
</div>"
`;

exports[`vue/UITextarea renders correctly - with helper 1`] = `
"<div class=\\"ui-textarea\\">
  <!---->
  <div class=\\"ui-textarea__wrapper\\"><textarea id=\\"z8a8d7a5zad4\\" name=\\"test\\" autocomplete=\\"on\\" class=\\"ui-textarea__wrapper__field\\"></textarea></div> <span class=\\"ui-textarea__helper\\"> Text </span>
</div>"
`;

exports[`vue/UITextarea renders correctly - with id 1`] = `
"<div id=\\"test\\" class=\\"ui-textarea\\">
  <!---->
  <div class=\\"ui-textarea__wrapper\\"><textarea id=\\"z8a8d7a5zad4\\" name=\\"test\\" autocomplete=\\"on\\" class=\\"ui-textarea__wrapper__field\\"></textarea></div>
  <!---->
</div>"
`;

exports[`vue/UITextarea renders correctly - with label 1`] = `
"<div class=\\"ui-textarea\\"><label for=\\"z8a8d7a5zad4\\" class=\\"ui-textarea__label\\"><strong>Text</strong></label>
  <div class=\\"ui-textarea__wrapper\\"><textarea id=\\"z8a8d7a5zad4\\" name=\\"test\\" autocomplete=\\"on\\" class=\\"ui-textarea__wrapper__field\\"></textarea></div>
  <!---->
</div>"
`;

exports[`vue/UITextarea renders correctly - with listeners 1`] = `
"<div class=\\"ui-textarea\\">
  <!---->
  <div class=\\"ui-textarea__wrapper\\"><textarea id=\\"z8a8d7a5zad4\\" name=\\"test\\" autocomplete=\\"on\\" class=\\"ui-textarea__wrapper__field\\"></textarea></div>
  <!---->
</div>"
`;

exports[`vue/UITextarea renders correctly - with maxlength 1`] = `
"<div class=\\"ui-textarea\\">
  <!---->
  <div class=\\"ui-textarea__wrapper\\"><textarea id=\\"z8a8d7a5zad4\\" name=\\"test\\" maxlength=\\"10\\" autocomplete=\\"on\\" class=\\"ui-textarea__wrapper__field\\"></textarea></div>
  <!---->
</div>"
`;

exports[`vue/UITextarea renders correctly - with placeholder 1`] = `
"<div class=\\"ui-textarea\\">
  <!---->
  <div class=\\"ui-textarea__wrapper\\"><textarea id=\\"z8a8d7a5zad4\\" name=\\"test\\" placeholder=\\"text\\" autocomplete=\\"on\\" class=\\"ui-textarea__wrapper__field\\"></textarea></div>
  <!---->
</div>"
`;

exports[`vue/UITextarea renders correctly - with transform 1`] = `
"<div class=\\"ui-textarea\\">
  <!---->
  <div class=\\"ui-textarea__wrapper\\"><textarea id=\\"z8a8d7a5zad4\\" name=\\"test\\" maxlength=\\"10\\" autocomplete=\\"on\\" class=\\"ui-textarea__wrapper__field\\"></textarea></div>
  <!---->
</div>"
`;

exports[`vue/UITextarea renders correctly - with value 1`] = `
"<div class=\\"ui-textarea\\">
  <!---->
  <div class=\\"ui-textarea__wrapper\\"><textarea id=\\"z8a8d7a5zad4\\" name=\\"test\\" autocomplete=\\"on\\" class=\\"ui-textarea__wrapper__field\\"></textarea></div>
  <!---->
</div>"
`;
